tgui/packages/tgui/interfaces/SpawnPanel/CreateObjectSettings.tsx

```diff
@@ -0,0 +1,420 @@
+      setter: setSpawnLocation,
+    });
+    sendUpdatedSettings({ where_dropdown_value: value });
+  };
+
+  const updateDirection = (value: number) => {
+    setStateAndStorage({
+      value,
+      storageKey: 'spawnpanel-direction',
+      setter: setDirection,
+    });
+    sendUpdatedSettings({ dir: [1, 2, 4, 8][value] });
+  };
+
+  const updateObjectName = (value: string) => {
+    setStateAndStorage({
+      value,
+      storageKey: 'spawnpanel-object_name',
+      setter: setObjectName,
+    });
+    sendUpdatedSettings({ object_name: value });
+  };
+
+  const updateOffset = (value: string) => {
+    setStateAndStorage({
+      value,
+      storageKey: 'spawnpanel-offset',
+      setter: setOffset,
+    });
+    sendUpdatedSettings({ offset: value });
+  };
+
+  const sendUpdatedSettings = (
+    changedSettings: Partial<Record<string, unknown>> = {},
+  ) => {
+    const currentSettings = {
+      object_count: amount,
+      offset_type: cordsType ? 'Absolute offset' : 'Relative offset',
+      where_dropdown_value: spawnLocation,
+      dir: [1, 2, 4, 8][direction],
+      offset: offset,
+      object_name: objectName,
+      custom_icon: iconSettings.icon,
+      custom_icon_state: iconSettings.iconState,
+      custom_icon_size: iconSettings.iconSize,
+      ...changedSettings,
+    };
+    act('update-settings', currentSettings);
+  };
+
+  useEffect(() => {
+    const loadStoredValues = async () => {
+      const storedAmount = await storage.get('spawnpanel-object_count');
+      const storedCordsType = await storage.get('spawnpanel-offset_type');
+      const storedSpawnLocation = await storage.get(
+        'spawnpanel-where_dropdown_value',
+      );
+      const storedDirection = await storage.get('spawnpanel-direction');
+      const storedObjectName = await storage.get('spawnpanel-object_name');
+      const storedOffset = await storage.get('spawnpanel-offset');
+
+      if (storedAmount) setAmount(storedAmount);
+      if (storedCordsType !== undefined) setCordsType(storedCordsType);
+      if (storedSpawnLocation) setSpawnLocation(storedSpawnLocation);
+      if (storedDirection !== undefined) setDirection(storedDirection);
+      if (storedObjectName !== undefined) setObjectName(storedObjectName);
+      if (storedOffset !== undefined) setOffset(storedOffset);
+    };
+
+    loadStoredValues();
+  }, []);
+
+  const isTargetMode =
+    spawnLocation === 'Targeted location' ||
+    spawnLocation === 'Targeted location (droppod)' ||
+    spawnLocation === "In targeted mob's hand";
+
+  const isPreciseModeActive = data?.precise_mode === 'Target';
+  const isMarkModeActive = data?.precise_mode === 'Mark';
+  const isCopyModeActive = data?.precise_mode === 'Copy';
+
+  const disablePreciseMode = function (): void {
+    if (isPreciseModeActive) {
+      act('toggle-precise-mode', {
+        newPreciseType: 'Off',
+      });
+    }
+  };
+
+  const handleSpawn = function (): void {
```

Comment:
old js functions?
please make these consistent with the rest of the arrow functions